### ISSUE TITLE
OIDC Authentication failing after upgrading to 1.1.1

### ISSUE THREAD

**Issue by edsonmarquezani** [argoproj/argo-cd#2047]
In 1.0.2, I was able to authenticate successfully against my own OIDC provider (Keycloak) with the following configuration in `argocd-cm` configmap.

```
data:
  oidc.config: |
    clientID: argo-cd
    clientSecret: <secret>
    issuer: https://<address>/auth/realms/prod
    name: SSO
``` 

After upgrading to 1.1.1, `argo-server` fails at start with the following error output:

```
time="2019-07-29T18:49:13Z" level=info msg="Starting configmap/secret informers"
time="2019-07-29T18:49:14Z" level=info msg="Configmap/secret informer synced"
time="2019-07-29T18:49:14Z" level=info msg="Starting configmap/secret informers"
time="2019-07-29T18:49:14Z" level=info msg="configmap informer cancelled"
time="2019-07-29T18:49:14Z" level=info msg="secrets informer cancelled"
time="2019-07-29T18:49:14Z" level=info msg="Configmap/secret informer synced"
2019/07/29 18:49:14 [Model:]
2019/07/29 18:49:14 r.r: sub, res, act, obj
2019/07/29 18:49:14 p.p: sub, res, act, obj, eft
2019/07/29 18:49:14 e.e: some(where (p_eft == allow)) && !some(where (p_eft == deny))
2019/07/29 18:49:14 m.m: g(r_sub, p_sub) && keyMatch(r_res, p_res) && keyMatch(r_act, p_act) && keyMatch(r_obj, p_obj)
2019/07/29 18:49:14 g.g: _, _
2019/07/29 18:49:14 [Policy:]
2019/07/29 18:49:14 [p :  sub, res, act, obj, eft :  []]
2019/07/29 18:49:14 [g :  _, _ :  []]
2019/07/29 18:49:14 [Role links for: g]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x171a126]

goroutine 1 [running]:
github.com/argoproj/argo-cd/server.(*ArgoCDServer).registerDexHandlers(0xc0008c60f0, 0xc0008180f0)
	/go/src/github.com/argoproj/argo-cd/server/server.go:560 +0xa6
github.com/argoproj/argo-cd/server.(*ArgoCDServer).newHTTPServer(0xc0008c60f0, 0x1e5d020, 0xc00078a980, 0x1f90, 0x1e35200, 0xc0008180c0, 0x40b901)
	/go/src/github.com/argoproj/argo-cd/server/server.go:535 +0xa37
github.com/argoproj/argo-cd/server.(*ArgoCDServer).Run(0xc0008c60f0, 0x1e5d020, 0xc00078a980, 0x1f90, 0x1f93)
	/go/src/github.com/argoproj/argo-cd/server/server.go:210 +0xcf
github.com/argoproj/argo-cd/cmd/argocd-server/commands.NewCommand.func1(0xc0005f7b00, 0xc0001dad50, 0x0, 0x3)
	/go/src/github.com/argoproj/argo-cd/cmd/argocd-server/commands/root.go:89 +0x52c
github.com/spf13/cobra.(*Command).execute(0xc0005f7b00, 0xc0000380d0, 0x3, 0x3, 0xc0005f7b00, 0xc0000380d0)
	/go/src/github.com/spf13/cobra/command.go:702 +0x2d3
github.com/spf13/cobra.(*Command).ExecuteC(0xc0005f7b00, 0xc0004adf60, 0xc0001642b0, 0xc0004adf80)
	/go/src/github.com/spf13/cobra/command.go:783 +0x2dc
github.com/spf13/cobra.(*Command).Execute(0xc0005f7b00, 0xc0006c1f88, 0xc0000440b8)
	/go/src/github.com/spf13/cobra/command.go:736 +0x2b
main.main()
	/go/src/github.com/argoproj/argo-cd/cmd/argocd-server/main.go:14 +0x27
```

I don't have any Dex connector configuration. Let me know what further information would help to debug this.

**Comment by alexec** [comment]
@jannfis do you think this is fixed by you recent change?

**Comment by jannfis** [comment]
@alexc Nope, don't think this specific error has to do something with the build tool `go build` vs `packr build` used (if you refer to #2041 that is). Must be something else.

**Comment by jannfis** [comment]
@alexec I think the bug is fixed in `master` anyway, line 560 of 1.1 server.go is:
```
559:	tlsConfig := a.settings.TLSConfig()
560:	tlsConfig.InsecureSkipVerify = true
```
I guess a.settings.TLSConfig() returns nil, and therefore 560 is a nil dereference. This commit should have fixed it: https://github.com/argoproj/argo-cd/commit/4dc959f3e579e0c30889d61eba14534df91eb251

**Comment by jannfis** [comment]
Might be worth a backport to `release-1.1` branch?

**Comment by jannfis** [comment]
@edsonmarquezani As a work around until a bug fix for 1.1 is around, I think you can enable TLS termination on argocd-server, if that's possible in your setup. The above nil pointer dereference crash should only happen when you have no TLS termination configured. Can you confirm this?

**Comment by alexmt** [comment]
Cherry-picked it into 1.1 . Looks like we already have enough for 1.1.2 patch release .

**Comment by edsonmarquezani** [comment]
@jannfis Yes, indeed, I'm running the server with `--insecure`. I can wait for the next release, no problems (I'm still in PoC stage with Argo). Thanks for the quick reply. ;)

**Comment by jannfis** [comment]
@edsonmarquezani The fix is in 1.1.2, released yesterday

**Comment by edsonmarquezani** [comment]
@jannfis Yes, I confirm the solution, just installed it, it's working now. Thanks again!
